aws/src/main/java/org/apache/iceberg/aws/RESTSigV4AuthManager.java

```diff
@@ -61,20 +61,29 @@ public RESTSigV4AuthSession catalogSession(
   }
 
   @Override
-  public AuthSession contextualSession(SessionCatalog.SessionContext context, AuthSession parent) {
+  public RESTSigV4AuthSession contextualSession(
+      SessionCatalog.SessionContext context, AuthSession parent) {
     AwsProperties contextProperties =
-        new AwsProperties(RESTUtil.merge(catalogProperties, context.properties()));
+        new AwsProperties(
+            RESTUtil.merge(
+                catalogProperties,
+                // Use both context properties and credentials to create the AwsProperties instance
```

Comment:
Uncovered this while adding more tests: the context credentials were being ignored, only the context properties were being used – but some information might be present in the credentials map, such as AWS access key ids + secrets.